### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by AndresCdo**
This merge request aims to improve the documentation of the functions in the project. The current documentation for the functions is either incomplete or lacks clarity, making it difficult for users to understand the functionalities of the functions.

### CODE REVIEW

**Comment by AndresCdo** [comment]
Hi, @Torantulino I believe it's important to increase documentation in the code and set type of variables of functions parameters.

**Comment by Torantulino** [comment]
Hi @AndresCdo, thank you for your contribution!
Please fix merge conflicts and I'll get to reviewing this!

**Comment by AndresCdo** [comment]
> Hi @AndresCdo, thank you for your contribution! Please fix merge conflicts and I'll get to reviewing this!

Conflicts were solved.

**Review by rmcsharry — APPROVED** [review]
LGTM, nice additions.

**Comment by AndresCdo** [comment]
@Torantulino Everything is ok now.

**Comment by caretak3r** [comment]
So all you did was clone, and copilot complete for each function? Man what happened to good PRs?

**Comment by AndresCdo** [comment]
> So all you did was clone, and copilot complete for each function? Man what happened to good PRs?

I did append values in the Dockerfile and I'm working on some implementation of Issues. this is my first time on an Opensource project, thank you for your empathy.

**Comment by ryanpeach** [comment]
Honestly @AndresCdo I think this is a valuable first step for this codebase. I'd do more than docstrings, but I have no problem if 90% of the comments are AI generated. More comments and code separation are desperately needed. You have my upvote.

**Comment by AndresCdo** [comment]
@Torantulino @rmcsharry  everything is ok now.

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: This change is unrelated to the documentation. Remove it and submit as a separate PR

File: Dockerfile
Comment (nponeccop) [inline Dockerfile:1]: Unrelated as well

File: scripts/ai_config.py [hunk 1]
```diff
@@ -26,16 +29,21 @@ def load(cls, config_file=SAVE_FILE):
 
         return cls(ai_name, ai_role, ai_goals)
 
+
```
Comment (nponeccop) [inline scripts/ai_config.py]: Avoid whitespace changes

File: scripts/ai_config.py [hunk 2]
```diff
@@ -26,16 +29,21 @@ def load(cls, config_file=SAVE_FILE):
...
             yaml.dump(config, file)
 
+
     def construct_full_prompt(self):
+        """Construct the full prompt for the AI to use."""
         prompt_start = """Your decisions must always be made independently without seeking user assistance. Play to your strengths as an LLM and pursue simple strategies with no legal complications."""
 
         # Construct full prompt
         full_prompt = f"You are {self.ai_name}, {self.ai_role}\n{prompt_start}\n\nGOALS:\n\n"
+
```
Comment (nponeccop) [inline scripts/ai_config.py]: Whitespace

File: scripts/ai_functions.py [hunk 3]
```diff
@@ -1,13 +1,13 @@
 from typing import List, Optional
-import json
+from json import dumps
```
Comment (nponeccop) [inline scripts/ai_functions.py]: It doesn't look like adding documentation to me. Put in a separate PR.

File: scripts/ai_functions.py [hunk 4]
```diff
@@ -18,26 +18,25 @@ def evaluate_code(code: str) -> List[str]:
 
 
 # Improving code
-
```
Comment (nponeccop) [inline scripts/ai_functions.py:21]: Whitespace change again

File: scripts/ai_functions.py [hunk 5]
```diff
@@ -18,26 +18,25 @@ def evaluate_code(code: str) -> List[str]:
...
+    args = [dumps(suggestions), code]
     description_string = """Improves the provided code based on the suggestions provided, making no other changes."""
 
     result_string = call_ai_function(function_string, args, description_string)
     return result_string
 
 
 # Writing tests
-
-
```
Comment (nponeccop) [inline scripts/ai_functions.py:35]: Whitespace

File: scripts/ai_functions.py [hunk 6]
```diff
@@ -18,26 +18,25 @@ def evaluate_code(code: str) -> List[str]:
...
 def write_tests(code: str, focus: List[str]) -> str:
+    """Generates test cases for the existing code, focusing on specific areas if required."""
     function_string = (
         "def create_test_cases(code: str, focus: Optional[str] = None) -> str:"
     )
-    args = [code, json.dumps(focus)]
```
Comment (nponeccop) [inline scripts/ai_functions.py]: The JSON correction above. Move apart

File: scripts/browse.py
Comment (nponeccop) [inline scripts/browse.py:1]: Again whitespace and unrelated changes

File: scripts/call_ai_function.py
Comment (nponeccop) [inline scripts/call_ai_function.py:1]: The import change is unrelated

File: scripts/chat.py
Comment (nponeccop) [inline scripts/chat.py:1]: The cfg movement is unrelated. Import and whitespace changes.

File: scripts/config.py
Comment (nponeccop) [inline scripts/config.py:1]: Whitespace.

File: scripts/main.py [hunk 7]
```diff
@@ -159,18 +177,21 @@ def load_variables(config_file="config.yaml"):
...
         should_continue = input(f"""Continue with the last settings? 
-Name:  {config.ai_name}
-Role:  {config.ai_role}
-Goals: {config.ai_goals}  
-Continue (y/n): """)
+            Name:  {config.ai_name}
+            Role:  {config.ai_role}
+            Goals: {config.ai_goals}  
+            Continue (y/n): """)
+
```
Comment (nponeccop) [inline scripts/main.py]: You got the idea at this point I guess

**Review by nponeccop — CHANGES_REQUESTED** [review]
Since this PR changes many files, make it pinpointed: JUST add documentation. It will simplify the merge. Note that I'm a layperson, it's @Torantulino who approves or disapproves commits.

**Comment by AndresCdo** [comment]
> Since this PR changes many files, make it pinpointed: JUST add documentation. It will simplify the merge. Note that I'm a layperson, it's @Torantulino who approves or disapproves commits.

Thank you so much for your Code Review, corrections were done.

File: scripts/ai_config.py [hunk 8]
```diff
@@ -27,15 +29,18 @@ def load(cls, config_file=SAVE_FILE):
...
             yaml.dump(config, file)
 
+
     def construct_full_prompt(self):
+        """Construct the full prompt for the AI to use."""
         prompt_start = """Your decisions must always be made independently without seeking user assistance. Play to your strengths as an LLM and pursue simple strategies with no legal complications."""
 
         # Construct full prompt
-        full_prompt = f"You are {self.ai_name}, {self.ai_role}\n{prompt_start}\n\nGOALS:\n\n"
+        full_prompt = f"You are {self.ai_name}, {self.ai_role}\n{prompt_start}\n\nGOALS:\n\n"
```
Comment (nponeccop) [inline scripts/ai_config.py]: Remove the whitespace change here

File: scripts/main.py [hunk 9]
```diff
@@ -170,7 +173,7 @@ def construct_prompt():
 Name:  {config.ai_name}
 Role:  {config.ai_role}
 Goals: {config.ai_goals}  
-Continue (y/n): """)
```
Comment (nponeccop) [inline scripts/main.py]: Whitespace again

File: scripts/main.py [hunk 10]
```diff
@@ -223,7 +227,7 @@ def prompt_user():
     print_to_console(
         "Enter up to 5 goals for your AI: ",
         Fore.GREEN,
-        "For example: \nIncrease net worth, Grow Twitter Account, Develop and manage multiple businesses autonomously'")
```
Comment (nponeccop) [inline scripts/main.py]: Whitespace

File: scripts/main.py [hunk 11]
```diff
@@ -323,7 +328,7 @@ def parse_arguments():
             f"Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for {ai_name}...",
             flush=True)
         while True:
-            console_input = input(Fore.MAGENTA + "Input:" + Style.RESET_ALL)
```
Comment (nponeccop) [inline scripts/main.py]: Whitespace

**Review by nponeccop — CHANGES_REQUESTED** [review]
Remove the introduced trailing whitespace. You may also want to avoid adding and removing empty lines, but it's optional and PR is already pretty focused. Good job!

**Comment by AndresCdo** [comment]
> Remove the introduced trailing whitespace. You may also want to avoid adding and removing empty lines, but it's optional and PR is already pretty focused. Good job!

@Torantulino everything is alright! Let's merge it then.

**Review by nponeccop — DISMISSED** [review]

**Comment by AndresCdo** [comment]
#137

**Review by nponeccop — DISMISSED** [review]
It's not ideal, but good enough. Some whitespace changes creeped in, but they are in the same area that this commit changes, I think it's ok especially because #590 takes so long to merge.

File: scripts/speak.py [hunk 12]
```diff
@@ -15,6 +15,7 @@
 }
 
 def eleven_labs_speech(text, voice_index=0):
+    """Return the results of a google search"""
```
Comment (Torantulino) [inline scripts/speak.py]:
```suggestion
    """Speak text using elevenlabs.io's API"""
```

**Review by Torantulino — CHANGES_REQUESTED** [review]

**Review by Torantulino — APPROVED** [review]

**Comment by nponeccop** [comment]
@AndresCdo Happy end!

**Pull request merged**